Encode requests url's in `rucio.js`

maany opened the pull request:
Fix #7724 

This pull request improves the handling of URLs in the `RucioClient` JavaScript module by ensuring proper encoding of dynamic URL components (`scope` and `name`) using `encodeURIComponent`. 

### URL Encoding Improvements:

* Updated all API calls in `RucioClient` to use `encodeURIComponent` for `scope` and `name` parameters, ensuring proper encoding of special characters. This change affects multiple methods, including:
  - `scope_list`
  - `list_subscriptions`
  - `list_replication_rules`
  - `did_get_metadata`, `did_get_generic_metadata`, `did_get_files`, and `list_parent_dids` [[1]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L767-R767) [[2]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L788-R788) [[3]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L809-R809) [[4]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L828-R828)
  - `list_contents`, `did_get_rules`, and `get_request_by_did` [[1]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L847-R847) [[2]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L904-R904) [[3]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1047-R1047)
  - `update_subscription` and `create_subscription` [[1]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1086-R1086) [[2]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1106-R1106)
  - `get_did`, `set_metadata`, and `set_status` [[1]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1348-R1348) [[2]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1417-R1417) [[3]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1438-R1438)
  - `list_dataset_replicas`, `add_did`, `attach_dids`, and `detach_dids` [[1]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1502-R1502) [[2]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1542-R1542) [[3]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1565-R1565) [[4]](diffhunk://#diff-57eb2a50b4fca5d9bb30f87eccebff890e8ebaa52c3721fd1a520246a105f183L1588-R1588)

maany committed:
fix(ui): encode requests in `rucio.js` #7724

The old UI has a single file called rucio.js that acts as a rucio HTTTP client. It makes requests to the UI's /proxy endpoint, which relays the request to the configured rucio server.

What Luis introduced in his containers PR rucio/containers#402 , is way to to set the AllowEncodedSlashes directive in the apache that sits in front of the UI.

As a second step, we need to encode the URLs in the rucio.js and send it to the UI Proxy, which can forward it to the rucio server.

This commit is focused on the latter

rdimaio commented:
Checked with Mayank, converting it to draft as it should not be merged for now

labkode commented:
@rdimaio are there any side effects why cannot be merged?

rdimaio commented:
> @rdimaio are there any side effects why cannot be merged?

@maany mentioned it's best to not merge this for now, I believe he's working on some related changes but he can expand on this

maany committed:
fix(ui): correct URL construction in RucioClient methods #7725

Fixes URL construction in RucioClient methods when DIDs contain forward
slashes. The UI now properly encodes URLs before sending requests to
handle different Apache configurations:

- CERN IT uses `AllowEncodedSlashes NoDecode`, requiring client-side encoding
- CMS uses `AllowEncodedSlashes On`, where httpd handles decoding
- Without proper encoding, DIDs with '/' cause upstream function failures

maany commented:
Corrected URL formatting by ensuring '/' separator is present between 
encoded scope and encoded name parameters in DID-related endpoints.

Ready for merge after CI run passes.
